File: LifeFlow/DonationService/BloodCenter/BloodCenter.cs
using System.ComponentModel.DataAnnotations;
using DonationService.Commons;

namespace DonationService.BloodCenter;

public class BloodCenter: BaseEntity
{
    [MaxLength(256)]
    public string Name { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public int UnitsCapacity { get; set; }
    public int RBCUnits { get; set; }
    public int PlateletsUnits { get; set; }
    public int PlasmaUnits { get; set; }
    public bool IsCentralReserve { get; set; }
    public int SlotsCapacity { get; set; }
    public int? AddressId { get; set; }
    public TimeSpan OpenByTime { get; set; } = new(9, 0, 0); // 9 Am
    public TimeSpan CloseByTime { get; set; } = new(21, 0, 0); // 9 PM
}

File: LifeFlow/life-flow-web/src/types/BloodCenter.ts
import {TimeSpan} from "./TimeSpanAsserts.ts";

export interface BloodCenterApiResponse {
  id: number;
  name: string;
  latitude: string;
  longitude: string;
  unitsCapacity: number;
  rbcUnits: number;
  plateletsUnits: number;
  plasmaUnits: number;
  isCentralReserve: boolean;
  slotsCapacity: number;
  openByTime: string;
  closeByTime: string;
  distance: number;
}

export interface BloodCenter {
  id: number;
  name: string;
  distance: string;
  rbc: number;
  capacity: number;
  openBy: TimeSpan;
  closeBy: TimeSpan;
  latitude: string;
  longitude: string;
}
